src/cli/agent.rs

```diff
@@ -37,23 +37,24 @@ pub(crate) fn run_agent() -> Result<()> {
     PROCESS_START_TIME.set(start_time.timestamp());
 
     trace!("creating actor system");
-    let sys = actix::System::builder()
-        .name(crate_name!())
-        .stop_on_panic(true)
-        .build();
+    let sys = actix::System::new();
 
-    trace!("creating metrics service");
-    let _metrics_addr = metrics::MetricsService::bind_socket()?.start();
+    sys.block_on(async {
```

Comment:
Self-note: API massaging here mostly caused by breaking changes caused by https://github.com/actix/actix-net/pull/207 in `actix-rt`.

Reply:
I'm not sure if that is the cause of further issues, but all the address variables that were previously scoped to `main` are now only alive inside this `async` block.

Reply:
That should be fine right, since they're moved into structs later.